plugins/tech-insights/src/api/TechInsightsClient.ts

```diff
@@ -45,76 +45,76 @@ export class TechInsightsClient implements TechInsightsApi {
+    const { namespace, kind, name } = entity;
+    const entityQuery = `entity=${encodeURIComponent(
+      kind.toLocaleLowerCase('en-US'),
+    )}:${encodeURIComponent(namespace)}/${encodeURIComponent(name)}`;
+    const idsQuery = facts.map(id => `ids[]=${id}`).join('&');
+    return await this.api<InsightFact>(
+      `/facts/latest?${entityQuery}&${idsQuery}`,
+    );
+  }
+
   getCheckResultRenderers(types: string[]): CheckResultRenderer[] {
     const renderers = this.renderers ?? [jsonRulesEngineCheckResultRenderer];
     return renderers.filter(d => types.includes(d.type));
   }
 
   async getAllChecks(): Promise<Check[]> {
-    const url = await this.discoveryApi.getBaseUrl('tech-insights');
-    const { token } = await this.identityApi.getCredentials();
-    const response = await fetch(`${url}/checks`, {
-      headers: token
-        ? {
-            Authorization: `Bearer ${token}`,
-          }
-        : undefined,
-    });
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
-    }
-    return await response.json();
+    return this.api('/checks');
   }
 
   async runChecks(
     entityParams: CompoundEntityRef,
     checks?: string[],
   ): Promise<CheckResult[]> {
-    const url = await this.discoveryApi.getBaseUrl('tech-insights');
-    const { token } = await this.identityApi.getCredentials();
     const { namespace, kind, name } = entityParams;
     const requestBody = { checks };
-    const response = await fetch(
-      `${url}/checks/run/${encodeURIComponent(namespace)}/${encodeURIComponent(
+    return this.api(
+      `/checks/run/${encodeURIComponent(namespace)}/${encodeURIComponent(
         kind,
       )}/${encodeURIComponent(name)}`,
       {
         method: 'POST',
         body: JSON.stringify(requestBody),
-        headers: {
-          'Content-Type': 'application/json',
-          ...(token && { Authorization: `Bearer ${token}` }),
-        },
       },
     );
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
-    }
-    return await response.json();
   }
 
   async runBulkChecks(
     entities: CompoundEntityRef[],
     checks?: Check[],
   ): Promise<BulkCheckResponse> {
-    const url = await this.discoveryApi.getBaseUrl('tech-insights');
-    const { token } = await this.identityApi.getCredentials();
     const checkIds = checks ? checks.map(check => check.id) : [];
     const requestBody = {
       entities,
       checks: checkIds.length > 0 ? checkIds : undefined,
     };
-    const response = await fetch(`${url}/checks/run`, {
+    return this.api('/checks/run', {
       method: 'POST',
       body: JSON.stringify(requestBody),
+    });
+  }
+
+  private async api<T>(path: string, init?: RequestInit): Promise<T> {
+    const url = await this.discoveryApi.getBaseUrl('tech-insights');
+    const { token } = await this.identityApi.getCredentials();
+
+    return fetch(`${url}${path}`, {
+      ...init,
       headers: {
```

Comment:
What if init already had headers? :) Well maybe that doesn't apply here right now, but still, a bit of a gotcha down the line.

```ts
const request = new Request(path, init);
if (!request.headers.has('content-type')) {
  request.headers.set('content-type', 'application/json');
}
if (token && !request.headers.has('authorization')) {
  request.headers.set('authorization', `Bearer ${token}`);
}

fetch(request).then(...
```

Another possible refactor (for the future, since it's breaking) is, if we inject `fetchApi` into this class then that will take care of filling in the authorization header transparently.